asapdiscovery-docking/asapdiscovery/docking/openeye.py

```diff
@@ -298,45 +316,104 @@ def _dock(
                             pose_res, retcode = self.run_oe_posit_docking(
                                 opts, pose_res, dus, lig_oemol, self.num_poses
                             )
-
-                    if retcode == oedocking.OEDockingReturnCode_Success:
-                        for result in pose_res.GetSinglePoseResults():
-                            posed_mol = result.GetPose()
-                            prob = result.GetProbability()
-
-                            posed_ligand = Ligand.from_oemol(
-                                posed_mol, **set.ligand.dict()
-                            )
-                            # set SD tags
-                            sd_data = {
-                                DockingResultCols.DOCKING_CONFIDENCE_POSIT.value: prob,
-                                DockingResultCols.POSIT_METHOD.value: oedocking.OEPositMethodGetName(
-                                    result.GetPositMethod()
-                                ),
-                            }
-                            posed_ligand.set_SD_data(sd_data)
-
-                            # Generate info about which target was actually used by multi-reference docking
-                            if isinstance(set, DockingInputMultiStructure):
-                                docked_target = set.complexes[result.GetReceptorIndex()]
-                                input_pair = DockingInputPair(
+                if retcode == oedocking.OEDockingReturnCode_Success:
+                    input_pairs = []
+                    posed_ligands = []
+                    for i, result in enumerate(pose_res.GetSinglePoseResults()):
+                        posed_mol = result.GetPose()
+                        prob = result.GetProbability()
+
+                        # clear SD data for uniformity of results
+                        from asapdiscovery.data.backend.openeye import clear_SD_data
```

Comment:
We use the tags that were assigned to the ligand on creationg downstream of docking sometimes, so I don't think we should be clearing the SD data here.

Reply:
We add back the tags when we create the posed ligand in line 331. The purpose of this is to remove the "Alternate Pose" tag that is present for all the poses after the first, which makes the length of that list smaller than the number of poses. I can update the comment to make it more obvious